wpical/src/main/native/cpp/fieldmap.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "fieldmap.h"
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+fieldmap::fieldmap() {}
+
+fieldmap::fieldmap(wpi::json json) {
+  for (auto& tag : json["tags"].items()) {
+    double tagXPos = tag.value()["pose"]["translation"]["x"];
+    double tagYPos = tag.value()["pose"]["translation"]["y"];
+    double tagZPos = tag.value()["pose"]["translation"]["z"];
+    double tagWQuat = tag.value()["pose"]["rotation"]["quaternion"]["W"];
+    double tagXQuat = tag.value()["pose"]["rotation"]["quaternion"]["X"];
+    double tagYQuat = tag.value()["pose"]["rotation"]["quaternion"]["Y"];
+    double tagZQuat = tag.value()["pose"]["rotation"]["quaternion"]["Z"];
+
+    tagVec.push_back(tag::pose(tagXPos, tagYPos, tagZPos, tagWQuat, tagXQuat,
+                               tagYQuat, tagZQuat));
+  }
+}
+
+tag::pose fieldmap::getTag(int tag) {
+  return tagVec[tag - 1];
+}
+
+int fieldmap::getNumTags() {
+  return tagVec.size();
+}
+
+double fieldmap::minimizeAngle(double angle) {
```

Comment:
This seems like a utility function?  We have a function in wpimath for this.

Reply:
linking wpimath with wpical seems to introduce issues with eigen, as we are using eigen from the ceres dependency. Am I going about this incorrectly?

Reply:
No, wpimath and ceres use totally different versions of Eigen. The apriltag library actually links to wpimath, but Gradle doesn't do transitive dependencies, and so we can kinda get away with it. The right way to do this is to make an overlay port in thirdparty-ceres that uses the same Eigen version we use, then we should be able to link everything without problems.

Reply:
For now, is it worth removing the angle wrapping to get this over the finish line?

Reply:
I think we can fix it later.

Reply:
Ok.  This should at least be made a static function though.